scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.UsersDBConstants;
+import com.bittercode.model.StoreException;
+import com.bittercode.model.User;
+import com.bittercode.model.UserRole;
+import com.bittercode.service.UserService;
+import com.bittercode.util.DBUtil;
+
+public class UserServiceImpl implements UserService {
+
+    private static final String registerUserQuery = "INSERT INTO " + UsersDBConstants.TABLE_USERS
+            + "  VALUES(?,?,?,?,?,?,?,?)";
+
+    private static final String loginUserQuery = "SELECT * FROM " + UsersDBConstants.TABLE_USERS + " WHERE "
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
+        try {
+            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Improve user type determination logic</b></div>
<div id="fix">

The user type is being determined based on the `UserRole`. Consider using an enum method or a more robust approach to map roles to user types, as the current implementation might not scale well if more roles are added in the future.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -30,7 +30,7 @@
          PreparedStatement ps;
          User user = null;
          try {
 -            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
 +            String userType = UserTypeMapper.getUserTypeString(role);
              ps = con.prepareStatement(loginUserQuery);
              ps.setString(1, email);
              ps.setString(2, password);
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2437361366>#e347cb</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged